shared/bundle_analysis/report.py

```diff
@@ -170,6 +171,73 @@ def ingest(self, path: str) -> int:
+        Rule 2. Previous and current asset shared the same set of module names
+        """
+        for curr_bundle_report in self.bundle_reports():
+            for prev_bundle_report in prev_bundle_analysis_report.bundle_reports():
+                if curr_bundle_report.name == prev_bundle_report.name:
+                    associated_assets_found = []
+
+                    # Rule 1 check
+                    prev_asset_hashed_names = {
+                        a.hashed_name: a.uuid
+                        for a in prev_bundle_report.asset_reports()
+                    }
+                    for curr_asset in curr_bundle_report.asset_reports():
+                        if curr_asset.asset_type == models.AssetType.JAVASCRIPT:
+                            if curr_asset.hashed_name in prev_asset_hashed_names:
+                                associated_assets_found.append(
+                                    [
+                                        prev_asset_hashed_names[curr_asset.hashed_name],
+                                        curr_asset.uuid,
+                                    ]
+                                )
+
+                    # Rule 2 check
+                    prev_module_asset_mapping = {}
+                    for prev_asset in prev_bundle_report.asset_reports():
+                        if prev_asset.asset_type == models.AssetType.JAVASCRIPT:
+                            prev_modules = tuple(
+                                sorted(
+                                    frozenset([m.name for m in prev_asset.modules()])
+                                )
+                            )
+                            # NOTE: Assume two non-related assets CANNOT have the same set of modules
+                            # though in reality there can be rare cases of this but we
+                            # will deal with that later if it becomes a prevalent problem
+                            prev_module_asset_mapping[prev_modules] = prev_asset.uuid
+
+                    for curr_asset in curr_bundle_report.asset_reports():
+                        if curr_asset.asset_type == models.AssetType.JAVASCRIPT:
+                            curr_modules = tuple(
+                                sorted(
+                                    frozenset([m.name for m in curr_asset.modules()])
+                                )
+                            )
+                            if curr_modules in prev_module_asset_mapping:
+                                associated_assets_found.append(
+                                    [
+                                        prev_module_asset_mapping[curr_modules],
+                                        curr_asset.uuid,
+                                    ]
+                                )
+
+                    # Update the Assets table for the bundle
+                    # TODO: Use SQLalchemy ORM to update instead of raw SQL
```

Comment:
Is there a ticket for this TODO already? Big chances of it being forgotten

Reply:
Added link to it.